### PR TITLE
#4005 For images in set  removed if user selected one image

### LINKED ISSUE #4005: "(For all images in set)" appears even when I am not uploading a set but rather a single image
One of the steps during upload asks us:
"(for all images in set)..."

Well, in this certain case there was only one media item. Please detect that and have the wording reflect that. Otherwise the user gets very worried that he's about to upload a big bunch of stuff that he didn't intend to.

In fact, showing an exact count, would make the user confident of what is about to happen.
2.13.1.818~9d99c25

### PR DESCRIPTION
**Pull request by gouri-panda**
Fixes #4005 

What changes did you make and why?
Created a variable for storing if the user selected multiple images or not in UploadActivity then it will access by the fragment
and then it will set the text depends on the variable.

### CODE REVIEW

**Comment by gouri-panda** [comment]
Ping @nicolas-raoul for review.

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
Looks good, only minor changes to perform.

File: app/src/main/java/fr/free/nrw/commons/upload/categories/UploadCategoriesFragment.java [hunk 1]
```diff
@@ -78,6 +84,16 @@ private void addTextChangeListenerToEtSearch() {
                 .observeOn(AndroidSchedulers.mainThread())
                 .subscribe(filter -> searchForCategory(filter.toString()), Timber::e);
     }
+    private void setTvSubTitle() {
+        Log.d("gouri", "setTvSubTitle: in uploadfragments");
```
[the remark this answers is not in the document]
Comment (gouri-panda) [reply app/src/main/java/fr/free/nrw/commons/upload/categories/UploadCategoriesFragment.java:88]: D'oh! sorry I was debugging something.

File: app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java [hunk 2]
```diff
@@ -102,6 +102,7 @@
     private Place place;
     private List<UploadableFile> uploadableFiles = Collections.emptyList();
     private int currentSelectedPosition = 0;
+    private boolean isMultipleItemSelected = false;
```
[the remark this answers is not in the document]
Comment (gouri-panda) [reply app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java:105]: ok

File: app/src/main/java/fr/free/nrw/commons/upload/categories/UploadCategoriesFragment.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (gouri-panda) [reply app/src/main/java/fr/free/nrw/commons/upload/categories/UploadCategoriesFragment.java]: done

File: app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (gouri-panda) [reply app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java]: done

**Comment by gouri-panda** [comment]
@nicolas-raoul Is there something wrong with PR?

**Comment by nicolas-raoul** [comment]
@gouri-panda It looks great but I have been super busy and almost no time for this app reecently :'-(

Could anyone test this PR? Thanks a lot!

**Comment by gouri-panda** [comment]
@nicolas-raoul  No rush take your time.

**Comment by nicolas-raoul** [comment]
@gouri-panda Very sorry for the delay, I now have time to review your PRs :-)
The change looks good, but would you mind rebasing on the latest master? Thanks!

**Comment by gouri-panda** [comment]
@nicolas-raoul sure!

**Comment by gouri-panda** [comment]
@nicolas-raoul @neslihanturan This PR is ready now.

**Comment by nicolas-raoul** [comment]
I tried compiling but could not: `MediaLicenseFragment.java:81: error: ';' expected`

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
Probably need to fix syntax?

**Review by nicolas-raoul — APPROVED** [review]
Tested with both a single picture and multiple pictures, it works as intended.
Thanks a lot!

File: app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java [hunk 3]
```diff
@@ -103,6 +103,7 @@
     private Place place;
     private List<UploadableFile> uploadableFiles = Collections.emptyList();
     private int currentSelectedPosition = 0;
+    private boolean isMultipleFilesSelected = false;
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java:106]: Ah, I forgot! We need javadoc for each new method/member.
After that I will merge.

File: app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java [hunk 4]
```diff
@@ -386,11 +387,15 @@ private void receiveInternalSharedItems() {
         Timber.d("Received intent %s with action %s", intent.toString(), intent.getAction());
 
         uploadableFiles = intent.getParcelableArrayListExtra(EXTRA_FILES);
+        isMultipleFilesSelected = uploadableFiles.size() > 1;
         Timber.i("Received multiple upload %s", uploadableFiles.size());
 
         place = intent.getParcelableExtra(PLACE_OBJECT);
         resetDirectPrefs();
     }
+    public boolean getIsMultipleFilesSelected() {
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java:396]: javadoc

File: app/src/main/java/fr/free/nrw/commons/upload/categories/UploadCategoriesFragment.java [hunk 5]
```diff
@@ -88,6 +94,15 @@ private void addTextChangeListenerToEtSearch() {
                 .observeOn(AndroidSchedulers.mainThread())
                 .subscribe(filter -> searchForCategory(filter.toString()), Timber::e);
     }
+    private void setTvSubTitle() {
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/categories/UploadCategoriesFragment.java:97]: javadoc, and also one empty line before the method, to make it easier to see.

File: app/src/main/java/fr/free/nrw/commons/upload/depicts/DepictsFragment.java [hunk 6]
```diff
@@ -86,6 +91,15 @@ public void onClick(View v) {
         initRecyclerView();
         addTextChangeListenerToSearchBox();
     }
+    private void setDepictsSubTitle() {
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/depicts/DepictsFragment.java:94]: javadoc and spacing

File: app/src/main/java/fr/free/nrw/commons/upload/license/MediaLicenseFragment.java [hunk 7]
```diff
@@ -84,6 +89,16 @@ public void onClick(View v) {
         presenter.getLicenses();
     }
 
+    private void setTvSubTitle() {
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/license/MediaLicenseFragment.java:92]: javadoc

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
Sorry a few minor things

File: app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java (hunk 3 above)
Comment (gouri-panda) [reply app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java:106]: @nicolas-raoul Should we add Javadoc here or getter fields?

Comment (nicolas-raoul) [reply app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java:106]: Both methods **and** members, actually. Sorry for the hassle, but documentation is usually written once, read many times, so it is a good investment :-)

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

**Comment by gouri-panda** [comment]
@nicolas-raoul Sorry for the delay. You can check now ;)

File: app/src/main/java/fr/free/nrw/commons/upload/depicts/DepictsFragment.java [hunk 8]
```diff
@@ -87,6 +92,19 @@ public void onClick(View v) {
         addTextChangeListenerToSearchBox();
     }
 
+    /**
+     * Sets the Depicts subtitle
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/upload/depicts/DepictsFragment.java:96]: Same here, could be more definitive. "Sets the Depict subtitle" is not actually what the method does. It decides the visibility of depics text view if multiplefile is not selecred as far as I can understand.

**Review by neslihanturan — CHANGES_REQUESTED** [review]
Some minor fixes on comments :)

File: app/src/main/java/fr/free/nrw/commons/upload/license/MediaLicenseFragment.java [hunk 9]
```diff
@@ -84,6 +89,19 @@ public void onClick(View v) {
         presenter.getLicenses();
     }
 
+    /**
+     * Sets the Tv Subtitle
```
[the remark this answers is not in the document]
Comment (gouri-panda) [reply app/src/main/java/fr/free/nrw/commons/upload/license/MediaLicenseFragment.java:93]: Thanks, @neslihanturan .From now on I'll keep this in my mind.

File: app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java (hunk 3 above)
Comment (nicolas-raoul) [reply app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java:107]: Javadoc must use this format for members too:
```
/**
 * Example javadoc
 */
```
Thanks!

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

File: app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java [hunk 10]
```diff
@@ -103,7 +105,9 @@
     private Place place;
     private List<UploadableFile> uploadableFiles = Collections.emptyList();
     private int currentSelectedPosition = 0;
-    // Checks for if multiple files selected
+    /*
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java:108]: Needs one more star here and another on the next line :-)
Must look like this:
```
/**
 * Example javadoc
 */
 ```
 Thanks!

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

File: app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java (hunk 10 above)
Comment (gouri-panda) [reply app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java:108]: Sorry about that!

**Review by neslihanturan — APPROVED** [review]

**Comment by neslihanturan** [comment]
Hey, this changes looks okay to me to merge, for some reason tests are failing. Do you have any estimation @gouri-panda ?

**Comment by gouri-panda** [comment]
@neslihanturan Thanks for reviewing files but I can't find build logs. Perhaps you can try with a new build tests. Thanks :)

**Comment by neslihanturan** [comment]
I can't see any point in this PR where Travis can fail. I can not see the build either. So merging this one for now to see if the build will fail or not. If so, we can always revert the commit.

**Pull request merged**